client/src/components/SearchResultsOffset.js

```diff
@@ -0,0 +1,167 @@
+import React from 'react'
+import { Box, Button, Paragraph, TextInput } from 'grommet'
+import { FormPrevious, FormNext } from 'grommet-icons'
+import { isOnlyNumbers } from '../helpers/isOnlyNumbers'
+import { useSearchResources } from '../hooks/useSearchResources'
+
+export const SearchResultsOffset = () => {
+  const {
+    query: { limit, offset },
+    response: { count },
+    setOffset,
+    goToSearchResults
+  } = useSearchResources()
+
+  const [enteredPageNumber, setEnteredPageNumber] = React.useState('')
+  const [
+    enteredPageNumberInRange,
+    setEnteredPageNumberInRange
+  ] = React.useState(false)
+
+  const safeOffset = parseInt(offset, 10) || 0
+  const safeCount = parseInt(count, 10) || 0
+  const safeLimit = parseInt(limit, 10)
+  const lastOffset =
+    safeCount && safeLimit > 0 ? Math.floor(safeCount / safeLimit) : 0
+
+  const startingPageSlices =
+    safeOffset <= 2 ? [0, 6] : [safeOffset - 2, safeOffset + 3]
+
+  const pageSlices = lastOffset - safeOffset <= 2 ? [-5] : startingPageSlices
+
+  const pageButtonsOffsets = [...Array(lastOffset + 1).keys()].slice(
+    ...pageSlices
+  )
+
+  const goToOffset = (page) => {
+    setOffset(page)
+    setEnteredPageNumber('')
+    goToSearchResults()
+  }
+
+  const atEnd = safeOffset === lastOffset
+  const atStart = safeOffset === 0
+
+  const handlePageNumberRequest = ({ target: { value } }) => {
+    const parsedValue = parseInt(value, 10)
+    if (isOnlyNumbers(value)) {
+      const pageOffset = parsedValue - 1
```

Comment:
This is because pages are 0 indexed but that's confusing for users?

Reply:
yeah you expect to see page `1` of results but that corresponds to an offset of `0`